src/Components/Home.jsx

```diff
@@ -1,43 +1,128 @@
 /** @format */
 
-import { ListGroup, DropdownButton, Dropdown } from 'react-bootstrap';
+import {
+	ListGroup,
+	DropdownButton,
+	Dropdown,
+	Button,
+	ButtonGroup,
+} from 'react-bootstrap';
 import { useEffect, useState } from 'react';
 
 import * as articlesApi from '../utils/articlesApi';
 
 import ArticleCard from './ArticleCard';
 import { fetchTopics } from '../utils/topicsApi';
+import { useNavigate } from 'react-router-dom';
 
 export default function Home() {
 	const [topics, setTopics] = useState([]);
+	const [sortBy, setSortBy] = useState(undefined);
 	const [filter, setFilter] = useState(undefined);
 	const [isLoading, setIsLoading] = useState(true);
 	const [articles, setArticles] = useState([]);
+	const [possibleSortBys, setPossibleSortBys] = useState([]);
+	const [order, setOrder] = useState('asc');
+
+	const navigate = useNavigate();
 
 	useEffect(() => {
 		setIsLoading(true);
 		fetchTopics().then((topicsFromApi) => {
-			setTopics(topicsFromApi.map((topicFromApi) => topicFromApi.slug))
-			setIsLoading(false)
-		})
-	},[]);
+			setTopics(topicsFromApi.map((topicFromApi) => topicFromApi.slug));
+			setIsLoading(false);
+		});
+	}, []);
 
 	useEffect(() => {
 		setIsLoading(true);
-		articlesApi.fetchArticles(filter).then((articlesFromApi) => {
+		articlesApi.fetchArticles(filter, sortBy, order).then((articlesFromApi) => {
 			setArticles(articlesFromApi);
+			setPossibleSortBys(Object.keys(articlesFromApi[0]));
```

Comment:
Odd use case here, I would suggest just hardcoding them to save on render time as this means your drop downs are dependent on a successful api call